File: dsa/Stack.java
// package dsa;

import java.util.Scanner;
import java.io.*;

/**
 * Node
 */
/**
 * Stack
 */
public class Stack {
  class Node {
    int data;
    Node next;
    public Node(int data) {
      this.data = data;
    }
  }

  public Node head;

  public void push(int value) {
    Node newNode = new Node(value);
    newNode.next = this.head;
    this.head = newNode;
    System.out.println("ok");
  }

  public void pop() {
    System.out.println(head.data);
    this.head = this.head.next;
  }

  public int back() {
    System.out.println(head.data);
    return head.data;
  }

  public int size() {
    int cnt = 0;
    Node tmp = this.head;
    while (tmp != null) {
      tmp = tmp.next;
      cnt = cnt + 1;
    }
    System.out.println(cnt);
    return cnt;
  }

  public void clear() {
    this.head = null;
    System.out.println("ok");
  }

  public static void main(String[] args) throws IOException{
    Stack stack = new Stack();
    StreamTokenizer tf = new StreamTokenizer(new BufferedReader(new InputStreamReader(System.in)));

    while (true) {
      tf.nextToken();
      if (tf.sval.equals("push")) {
        tf.nextToken();
        int value =  (int)tf.nval;
        stack.push(value);
      }
      else if (tf.sval.equals("pop")) {
        stack.pop();
      }
      else if (tf.sval.equals("clear")) {
        stack.clear();
      }
      else if (tf.sval.equals("size")) {
        stack.size();
      }
      else if (tf.sval.equals("back")) {
        stack.back();
      }
      else if (tf.sval.equals("exit")) {
        System.out.println("bye");
        break;
      }
    }
  }
    // scanner.close();
}

File: dsa/stack.cpp
/* #include <bits/stdc++.h> */
/* #include <string> */
/* using namespace std; */

/* class Node { */
/*   public: */
/*     int data; */
/*     Node * next; */
/*     Node(int data) { this->data = data; } */
/* }; */

/* std::vector<std::string> split (const std::string &s, char delim) { */
/*     std::vector<std::string> result; */
/*     std::stringstream ss (s); */
/*     std::string item; */

/*     while (getline (ss, item, delim)) { */
/*         result.push_back (item); */
/*     } */

/*     return result; */
/* } */

/* class Stack { */
/*   Node* head; */
/*   public: */
/*     Stack(); */

/*     void push(int data){ */
/*       Node* tmp = head; */
/*       while (tmp->next) */ 
/*         tmp = tmp->next; */
/*       Node * newnode = new Node(data); */
/*       tmp->next = newnode; */
/*     }; */

/*     void clear(){ */
/*       while (head) { */
/*         Node* tmp = head; */
/*         head = head->next; */
/*         delete tmp; */
/*       } */
/*     } */

/*     int size(){ */
/*       Node* tmp = head; */
/*       int size = 0; */
/*       while (tmp) { */
/*         size++; */
/*         tmp = tmp->next; */
/*       } */
/*       return size; */
/*     } */

/*     void pop(){ */
/*       Node* tmp = head; */
/*       if (!head->next) */ 
/*         head = nullptr; */
/*       while (tmp->next->next) */
/*         tmp = tmp->next; */
/*       std::cout << tmp->next->data << std::endl; */
/*       tmp->next = nullptr; */ 
/*     }; */

/*     Node* back(){ */
/*       Node* tmp = head; */
/*       while (tmp->next) */
/*         tmp = tmp->next; */
/*       cout << tmp->data; */
/*       return tmp; */
/*     }; */
/*     ~Stack() { */
/*       clear(); */
/*     }; */
/* }; */

/* int main(int argc, char *argv[]) { */
/*   string line; */

/*   while (getline(cin, line)) { */
/*     if (line == "exit") */ 
/*       std::cout << "bye" << std::endl; */
/*     if (line == "push") */ 
/*       std::cout << "bye" << std::endl; */
/*   } */
/*   return 0; */ 
/* } */
